components/common/GlobalNav.tsx

```diff
@@ -487,7 +488,17 @@ function GlobalNav(props) {
   );
   const hideLogoOnMobile = !!(theme.navTitleVisible && siblings.length);
 
-  const rootLink = plan.parent ? plan.parent?.viewUrl : '/';
+  const locale = useLocale();
+  let rootLink = '/';
+
+  if (plan.parent && plan.parent?.viewUrl) {
+    rootLink = plan.parent?.viewUrl;
+    const shouldAppendLocale = locale !== plan.primaryLanguage;
+    if (shouldAppendLocale) {
+      rootLink = `${plan.parent?.viewUrl}/${locale}/`;
+    }
+  }
```

Comment:
Nitpick: To improve readability, this could be moved to a dedicated `getRootLink` function. We also then wouldn't need to reassign `rootLink` but could just return the value based on the different if statements (in general we prefer not to use `let`)